packages/core/database/lib/index.js

```diff
@@ -58,9 +58,17 @@ class Database {
     return this.entityManager.getRepository(uid);
   }
 
-  getConnection(tableName) {
+  async transaction(cb) {
```

Comment:
If `transaction` return the promise, the client code could use it to return back to the caller.

`const entity = await strapi.db.transaction((em)=>{/* do some stuff */});`
```diff
async transaction(cb) {
-    await this.connection.transaction(async trx => {
+   return await this.connection.transaction(async trx => {  
      const em = createEntityManager(this, trx);
-      await cb(em);
+     return await cb(em); 
    });
  }

```